spec/lib/pact_broker/pacts/service_spec.rb

```diff
@@ -34,35 +34,116 @@ module Pacts
         let(:json_content) { { 'some' => 'json'}.to_json }
         let(:pact) { instance_double(PactBroker::Domain::Pact, json_content: json_content)}
 
-        before do
-          allow_any_instance_of(Pacts::Repository).to receive(:find_previous_pact).and_return(previous_pact)
+        subject { Service.pact_is_new_or_pact_has_changed_since_previous_version? pact }
+
+        context "when consumer version is untagged" do
+          before do
+            allow(pact).to receive(:consumer_version_tag_names).and_return([]);
+            allow_any_instance_of(Pacts::Repository).to receive(:find_previous_pact).with(pact, :untagged).and_return(previous_pact)
+          end
+
+          context "when a previous pact is found" do
+            let(:previous_pact) { instance_double(PactBroker::Domain::Pact, json_content: previous_json_content)}
+            let(:previous_json_content) { {'some' => 'json'}.to_json }
+
+            context "when the json_content is the same" do
+              it "returns false" do
+                expect(subject).to be_falsey
+              end
+            end
+
+            context "when the json_content is not the same" do
+              let(:previous_json_content) { {'some-other' => 'json'}.to_json }
+              it "returns truthy" do
+                expect(subject).to be_truthy
+              end
+            end
+          end
+
+          context "when a previous pact is not found" do
+            let(:previous_pact) { nil }
+
+            it "returns true" do
+              expect(subject).to be_truthy
+            end
+          end
         end
 
-        subject { Service.pact_is_new_or_pact_has_changed_since_previous_version? pact }
+        context "when consumer version has one tag" do
```

Comment:
Not sure of the value of this test when you have the "two tag" version below it. Seems redundant to me.

Reply:
The codebase is getting to the stage where I'm getting impatient for the tests to finish, so if there's any test that doesn't provide unique value, I reckon it's better out than in.

Reply:
Agreed. I wrote the tests whilst I was developing to validate the approach, but it is made redundant by the test below.